subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/ComponentDependencies.java

```diff
@@ -35,4 +36,18 @@
 
     void compileOnly(Object dependencyNotation);
     void compileOnly(Object dependencyNotation, Action<? super ExternalModuleDependency> configuration);
+
+    /**
+     * Creates a dependency on the API of the current version of Gradle.
+     *
+     * @return The dependency.
+     */
+    Dependency gradleApi();
+
+    /**
+     * Creates a dependency on the <a href="https://docs.gradle.org/current/userguide/test_kit.html" target="_top">Gradle test-kit</a> API.
+     *
+     * @return The dependency.
+     */
+    Dependency gradleTestKit();
```

Comment:
We don't want to add these to this interface if we can help it. See my other comment above where I think this is unnecessary since we are using the java-gradle-plugin.